app/controllers/users_controller.rb

```diff
@@ -1973,4 +2017,13 @@ def serialize_found_users(users)
 
     { users: ActiveModel::ArraySerializer.new(users, each_serializer: each_serializer).as_json }
   end
+
+  def find_unread_notifications_of_type(type, limit)
```

Comment:
Since this is pretty much a scope, I think we should just declare this as a scope on the `Notification` model.